### PR TITLE
Fix errors reported by phpstan

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

File: src/Datasource/RuleInvoker.php [hunk 1]
```diff
@@ -136,6 +136,7 @@ public function __invoke(EntityInterface $entity, array $scope): bool
             $entity->setInvalidField($errorField, $invalidValue);
         }
 
+        /** @phpstan-ignore-next-line */
```
Comment (othercorey) [inline src/Datasource/RuleInvoker.php:139]: Why did this change with a psalm version bump?

File: src/View/Widget/DateTimeWidget.php [hunk 2]
```diff
@@ -189,7 +189,7 @@ protected function formatDateTime($value, array $options): string
                 $dateTime = clone $value;
             } elseif (is_string($value) && !is_numeric($value)) {
                 $dateTime = new DateTime($value);
-            } elseif (is_int($value) || is_numeric($value)) {
+            } elseif (is_numeric($value)) {
```
Comment (othercorey) [inline src/View/Widget/DateTimeWidget.php:192]: This should really only accept integer strings, but whatever.

File: src/Datasource/RuleInvoker.php (hunk 1 above)
Comment (ADmad) [reply src/Datasource/RuleInvoker.php:139]: The error is caused by a new phpstan patch release, not psalm. I just chose to update psalm too while I was at it.

File: src/View/Widget/DateTimeWidget.php (hunk 2 above)
Comment (ADmad) [reply src/View/Widget/DateTimeWidget.php:192]: I agree, but it's not worth rocking the boat in 4.x, we can change it in 5.x.

**Pull request merged**